frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/usePopStateListener.ts

```diff
@@ -1,15 +1,23 @@
-import { updateActiveTableName, updateIsPopstateInProgress } from '@/stores'
+import type { ShowMode } from '@/schemas/showMode'
+import {
+  updateActiveTableName,
+  updateIsPopstateInProgress,
+  updateShowMode,
+} from '@/stores'
 import { useEffect } from 'react'
 
 export const usePopStateListener = () => {
   useEffect(() => {
     const handlePopState = async () => {
       const url = new URL(window.location.href)
       const tableName = url.searchParams.get('active')
+      const showMode = url.searchParams.get('showMode')
 
       updateIsPopstateInProgress(true)
       updateActiveTableName(tableName ?? undefined)
 
+      updateShowMode(showMode as ShowMode)
```

Comment:
The `as` here is dangerous. Because you can put any string in the query parameter.
It is a good to parse here with valibot, and if the value is invalid, put in the default value.

Reply:
Thank you for your feedback🙏. I've made the same adjustments to this one(https://github.com/liam-hq/liam/pull/452/commits/0890fafc470fb8f7501fdb7e3a14d838592fc8b5 ) as well.
https://github.com/liam-hq/liam/pull/452/commits/168010704d67685b61b7306768863717a0e3d072